dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/sip/netty/SipInboundChannelFactoryWs.java

```
@@ -0,0 +1,144 @@
+		}
+
+		String transport = chain.getTransport();
+
+		ListeningPoint lp = createListeningPoint(chain);//chain.getTransport(), chain.getChainName());
+
+		// create inbound channel,
+
+		String chainNumber = String.valueOf(s_chains++);
+		SipInboundChannel channel = null;
+		String outboundChainName = null;
+		
+		switch (chain.getType()) {
+		case udp:
+			outboundChainName = "UdpOutboundChain_" + chainNumber;
+			channel = SipUdpInboundChannel.instance(lp, outboundChainName);
+			break;
+		case tcp:
+			outboundChainName = "TcpOutboundChain_" + chainNumber;
+			channel = new SipTcpInboundChannel(lp, outboundChainName);
+			break;
+		case tls:
+		    outboundChainName = "TcpOutboundChain_" + chainNumber;
+		    channel = new SipTlsInboundChannel(lp, outboundChainName);
+			break;
+		}
+
+		try {
+			SIPConnectionFactoryImplWs.instance().addListeningConnection(lp, channel, chain.getChainName());
+			if (transport.equals(ListeningPoint.TRANSPORT_TCP)) {/*
+				ListeningPoint lpLB = new ListeningPointImpl("127.0.0.1", 5060, transport, chainName + "_lb");
+				SIPConnectionFactoryImplWs.instance().addListeningConnection(lpLB,
+						new SipTcpInboundChannel(lpLB, "TcpOutboundChain_lb_" + chainNumber), chainName + "_lb");*/
+			}
+			return lp;
+		} catch (IOException e) {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+				Tr.debug(tc, "failed creating a connection", e);
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * creates a jain listening point
+	 * 
+	 * @return a new listening point object, null on error
+	 */
+	private ListeningPoint createListeningPoint(GenericChain chain) {
+		String host = chain.getActiveHost();
+
+		if (host.equals("*")) {
+			host = "0.0.0.0";
+		}
+
+		return new ListeningPointImpl(host, chain.getActivePort(), chain.getTransport(), chain.getChainName());
+	}
+
+	@Override
+	public Channel findOrCreateChannel(ChannelData config) throws ChannelException {
+		// TODO Auto-generated method stub
```

Maybe these return null should throw unsupported op or something?

---

If this is same in the legacy code, we could get feedback from Tibor

---

It's same this way in chfw code.